### PR TITLE
resolve: use unixfs ResolveOnce

### PR DESCRIPTION
**Pull request by magik6k**
(This is assuming we want to make `ipfs resolve` operate on unixfs paths instead of ipld)

Fixes https://github.com/ipfs/go-ipfs/issues/5270#issuecomment-422112136

### CODE REVIEW

**Comment by magik6k** [comment]
Good point, I'll see how that works

**Comment by magik6k** [comment]
Done

File: core/coreapi/interface/options/name.go [hunk 1]
```diff
@@ -40,9 +43,12 @@ func NamePublishOptions(opts ...NamePublishOption) (*NamePublishSettings, error)
...
 func NameResolveOptions(opts ...NameResolveOption) (*NameResolveSettings, error) {
 	options := &NameResolveSettings{
-		Recursive: false,
-		Local:     false,
-		Cache:     true,
+		Depth: 1,
+		Local: false,
+		Cache: true,
+
+		DhtRecordCount: 16,
```
Comment (Stebalien) [inline core/coreapi/interface/options/name.go]: Not sure if we should hard-code this. Can we use the value from the DHT (unless we go with the "resolve options" solution in which case this point is moot).

**Comment by Stebalien** [comment]
Note: I'm fine keeping this as it is, I'd just like to consider a more generalized solution to adding more and more API options.

**Comment by Stebalien** [comment]
@magik6k This is wonderful. These issues have been plaguing us for *ages*.

**Pull request merged**